src/main/java/DavinciBot.java

```diff
@@ -0,0 +1,276 @@
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * DavinciBot is a simple bot that allows the user to manage a to-do list.
+ * The bot echoes commands entered by the user, adds tasks to the list,
+ * marks tasks as done, and unmarks tasks.
+ */
+public class DavinciBot {
+
+    private static final String LINE_SEPARATOR = "____________________________________________________________";
+    public static final int SPLIT_INTO_TWO_PARTS = 2;
+
+    /**
+     * Selects the icon corresponding to the type of task inputted by the user.
+     *
+     * @param taskArray Array of tasks.
+     */
+    private static void iconSelector(Task[] taskArray) {
+        for (int i = 0; i < taskArray.length; i++) {
+            String taskType = (taskArray[i] instanceof Deadline) ? "[D]" :
+                    (taskArray[i] instanceof Todo) ? "[T]" : "[E]";
+
+            System.out.println((i + 1) + ". " + taskType +
+                    " [" + taskArray[i].getStatusIcon() + "] " +
+                    taskArray[i].getDescription());
+        }
+    }
+    /**
+     * Displays the tasks in the user's to-do list.
+     *
+     * @param taskArray Array of tasks.
+     */
+    private static void displayTaskList(Task[] taskArray) {
+        if (taskArray.length == 0) {
+            System.out.println("No tasks entered yet.");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            iconSelector(taskArray);
+        }
+    }
+
+    /**
+     * Marks a specified task as completed.
+     *
+     * @param userInput User input specifying the task.
+     * @param taskArray Array of tasks.
+     */
+    private static void completeTask(String userInput, Task[] taskArray) {
+        try {
+            String[] parts = userInput.split(" ", SPLIT_INTO_TWO_PARTS);
+            if (parts.length > 1) {
+                int taskIndex = Integer.parseInt(parts[1]) - 1;
+                if (taskIndex >= 0 && taskIndex < taskArray.length) {
+                    taskArray[taskIndex].markAsDone();
+                    System.out.println(LINE_SEPARATOR);
+                    System.out.println("Nice job! I've marked this task as done :D");
+                    System.out.println("[" + taskArray[taskIndex].getStatusIcon() + "] " +
+                            taskArray[taskIndex].getDescription());
+                    System.out.println(LINE_SEPARATOR);
+                } else {
+                    throw new DavinciException("Invalid task index.");
+                }
```

Comment:
try to avoid deep nesting, you can abstract out the innermost if..else into a different function 

applies to the next 2 functions too

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting